File: tests-syntax/test-syntax.py
# Python Syntax Test File
# This file tests various syntax elements to verify theme colors

# ===== COMMENTS =====
# Single line comment
"""
Multi-line comment
with multiple lines
"""

# ===== KEYWORDS & CONTROL FLOW =====
name = "Evergarden"
version = 1.0
is_dark = True
is_enabled = False
none_value = None

if is_dark:
    print("Dark theme active")
elif not is_dark:
    print("Light theme active")
else:
    print("Unknown theme")

for i in range(10):
    print(f"Iteration {i}")

while condition:
    # do something
    pass

try:
    risky_operation()
except ValueError as error:
    print(f"Error occurred: {error}")
except (TypeError, AttributeError) as error:
    print(f"Multiple error types: {error}")
finally:
    print("Cleanup complete")

# ===== FUNCTIONS & DEFINITIONS =====
def greet(name: str) -> str:
    """Function with type hints and docstring"""
    return f"Hello, {name}"

def arrow_function(param1: int, param2: int) -> int:
    return param1 + param2

def *args_func(*args, **kwargs):
    print(f"Args: {args}")
    print(f"Kwargs: {kwargs}")

# ===== CLASSES =====
class ThemeManager:
    """Theme manager class with various methods"""
    
    def __init__(self, theme: str):
        self.current_theme = theme
        self.version = 1.0
        self._private_var = "private"
    
    @property
    def theme(self) -> str:
        return self.current_theme
    
    @theme.setter
    def theme(self, value: str):
        self.current_theme = value
    
    def get_theme(self) -> str:
        return self.current_theme
    
    def set_theme(self, theme: str) -> None:
        self.current_theme = theme
    
    @staticmethod
    def static_method():
        return "Static method"
    
    @classmethod
    def class_method(cls):
        return f"Class method of {cls.__name__}"

class ColorScheme(ThemeManager):
    """Inherited class"""
    
    def __init__(self, theme: str, colors: dict):
        super().__init__(theme)
        self.colors = colors

# ===== IMPORTS =====
import os
import sys
from typing import List, Dict, Optional, Union, Any
from dataclasses import dataclass
from enum import Enum
import json as js
from .local_module import local_function
from ..parent_module import parent_function

# ===== TYPES & TYPE HINTS =====
ThemeType = str
ColorHex = str
ThemeConfig = Dict[str, Any]
OptionalString = Optional[str]

# ===== ENUMS =====
class ThemeEnum(Enum):
    DARK = "dark"
    LIGHT = "light"
    AUTO = "auto"

# ===== DATA CLASSES =====
@dataclass
class ThemeData:
    name: str
    version: float
    colors: List[str]
    enabled: bool = True

# ===== STRINGS & NUMBERS =====
string_value = "This is a string"
raw_string = r"Raw string with \n no escape"
unicode_string = "Unicode string with ñiño"

template_string = f"Template string with {name} interpolation"
multiline_string = """
    Multiline
    string
    example
"""

integer = 42
float_val = 3.14159
hex_val = 0xFF0000
octal_val = 0o755
binary_val = 0b1010
complex_val = 3 + 4j

# ===== OPERATORS =====
sum_val = a + b
difference = a - b
product = a * b
quotient = a / b
floor_div = a // b
remainder = a % b
power = a ** b

is_equal = a == b
is_not_equal = a != b
is_greater = a > b
is_less_or_equal = a <= b

logical_and = a and b
logical_or = a or b
logical_not = not a

bitwise_and = a & b
bitwise_or = a | b
bitwise_xor = a ^ b
left_shift = a << b
right_shift = a >> b

# ===== LISTS, TUPLES, DICTS =====
numbers = [1, 2, 3, 4, 5]
nested_list = [[1, 2], [3, 4]]
list_comprehension = [x * 2 for x in numbers if x % 2 == 0]

coordinates = (10, 20)
named_tuple = (x=1, y=2, z=3)

theme_config = {
    "name": "Evergarden",
    "version": 1.0,
    "colors": {
        "background": "#0E1012",
        "foreground": "#e2e3e4"
    }
}

dict_comprehension = {x: x**2 for x in range(5)}

# ===== SETS =====
unique_numbers = {1, 2, 3, 4, 5}
set_comprehension = {x**2 for x in range(5)}

# ===== LAMBDA FUNCTIONS =====
square = lambda x: x * 2
add = lambda x, y: x + y

# ===== GENERATORS & ITERATORS =====
def number_generator():
    for i in range(10):
        yield i

gen_exp = (x * 2 for x in range(5))

# ===== CONTEXT MANAGERS =====
with open("file.txt", "r") as f:
    content = f.read()

# ===== ASYNC/AWAIT =====
import asyncio

async def fetch_data(url: str) -> dict:
    try:
        response = await asyncio.get_event_loop().run_in_executor(
            None, lambda: {"data": "mock"}
        )
        return response
    except Exception as error:
        print(f"Fetch error: {error}")
        raise

async def main():
    data = await fetch_data("https://example.com")
    print(data)

# ===== DECORATORS =====
def timer(func):
    def wrapper(*args, **kwargs):
        import time
        start = time.time()
        result = func(*args, **kwargs)
        end = time.time()
        print(f"{func.__name__} took {end - start} seconds")
        return result
    return wrapper

@timer
def slow_function():
    import time
    time.sleep(0.1)

# ===== REGULAR EXPRESSIONS =====
import re
pattern = r"test[0-9]+"
regex = re.compile(pattern, re.IGNORECASE | re.MULTILINE)

# ===== EXCEPTIONS =====
class ThemeError(Exception):
    """Custom exception for theme errors"""
    pass

class ColorError(ThemeError):
    """Custom exception for color errors"""
    pass

# ===== MODULES & PACKAGES =====
if __name__ == "__main__":
    print("Running as main module")

# ===== BUILT-IN FUNCTIONS =====
print(len(numbers))
print(max(numbers))
print(min(numbers))
print(sum(numbers))
print(sorted(numbers))
print(reversed(numbers))
print(enumerate(numbers))
print(zip(numbers, numbers))

# ===== TYPE CHECKING =====
isinstance(name, str)
issubclass(ColorScheme, ThemeManager)
hasattr(theme_config, "get")
callable(greet)

# ===== ATTRIBUTES =====
getattr(theme_config, "name", "default")
setattr(theme_config, "new_key", "new_value")
delattr(theme_config, "new_key")

# ===== SLICING =====
numbers[0:5]
numbers[::2]
numbers[::-1]

# ===== BOOLEAN OPERATIONS =====
bool_value = bool(0)
all_values = all([True, False, True])
any_values = any([True, False, True])

# ===== MATH OPERATIONS =====
import math
math.sqrt(16)
math.pow(2, 3)
math.pi
math.e

File: tests-syntax/test-syntax.js
// JavaScript/TypeScript Syntax Test File
// This file tests various syntax elements to verify theme colors

// ===== COMMENTS =====
// Single line comment
/* Multi-line comment */
/**
 * JSDoc comment with @param and @return
 * @param {string} name - The name parameter
 * @returns {string} - A greeting
 */

// ===== KEYWORDS & CONTROL FLOW =====
const name = "Evergarden";
let version = 1.0;
var isDark = true;

if (isDark) {
    console.log("Dark theme active");
} else if (!isDark) {
    console.log("Light theme active");
} else {
    console.log("Unknown theme");
}

for (let i = 0; i < 10; i++) {
    console.log(`Iteration ${i}`);
}

while (condition) {
    // do something
}

do {
    // something
} while (condition);

switch (theme) {
    case "dark":
        console.log("Dark");
        break;
    case "light":
        console.log("Light");
        break;
    default:
        console.log("Default");
}

// ===== FUNCTIONS & ARROWS =====
function greet(name) {
    return `Hello, ${name}`;
}

const arrowFunction = (param1, param2) => {
    return param1 + param2;
};

const shortArrow = x => x * 2;

// ===== CLASSES & INTERFACES =====
class ThemeManager {
    private currentTheme: string;
    protected version: number;
    
    constructor(theme: string) {
        this.currentTheme = theme;
        this.version = 1.0;
    }
    
    public getTheme(): string {
        return this.currentTheme;
    }
    
    public setTheme(theme: string): void {
        this.currentTheme = theme;
    }
}

interface ITheme {
    name: string;
    colors: string[];
}

interface IConfigurable {
    configure(options: any): void;
}

// ===== TYPES & TYPEDEFINITIONS =====
type ThemeType = "dark" | "light" | "auto";
type ColorHex = string;
type ThemeConfig = {
    name: string;
    background: ColorHex;
    foreground: ColorHex;
};

enum ThemeEnum {
    DARK = "dark",
    LIGHT = "light",
    AUTO = "auto"
}

// ===== STRINGS & NUMBERS =====
const stringValue = "This is a string";
const templateString = `Template string with ${name} interpolation`;
const multilineString = `
    Multiline
    string
    example
`;

const integer = 42;
const float = 3.14159;
const hex = 0xFF0000;
const binary = 0b1010;
const octal = 0o755;
const bigInt = 9007199254740991n;

// ===== OPERATORS =====
const sum = a + b;
const difference = a - b;
const product = a * b;
const quotient = a / b;
const remainder = a % b;

const isEqual = a === b;
const isNotEqual = a !== b;
const isGreater = a > b;
const isLessOrEqual = a <= b;

const logicalAnd = a && b;
const logicalOr = a || b;
const logicalNot = !a;

const bitwiseAnd = a & b;
const bitwiseOr = a | b;
const bitwiseXor = a ^ b;
const leftShift = a << b;
const rightShift = a >> b;

// ===== REGULAR EXPRESSIONS =====
const regex = /pattern/g;
const regexWithFlags = /test[0-9]+/gi;
const regexConstructor = new RegExp("pattern", "flags");

// ===== IMPORTS & EXPORTS =====
import { useState, useEffect } from "react";
import * as fs from "fs";
import { ThemeManager as TM } from "./theme-manager";
import type { ThemeType } from "./types";

export default ThemeManager;
export { ThemeManager, ITheme };
export type { ThemeConfig };

// ===== DECORATORS & METADATA =====
@sealed
class DecoratorExample {
    @log
    method() {
        console.log("Method called");
    }
}

// ===== ERROR HANDLING =====
try {
    riskyOperation();
} catch (error) {
    console.error("Error occurred:", error);
} finally {
    console.log("Cleanup complete");
}

throw new Error("Something went wrong");

// ===== ASYNC/AWAIT =====
async function fetchData(url: string): Promise<any> {
    try {
        const response = await fetch(url);
        const data = await response.json();
        return data;
    } catch (error) {
        console.error("Fetch error:", error);
        throw error;
    }
}

// ===== GENERATORS =====
function* numberGenerator() {
    let i = 0;
    while (true) {
        yield i++;
    }
}

// ===== MODULES & NAMESPACES =====
namespace ThemeNamespace {
    export const VERSION = "1.0.0";
    
    export class Util {
        static formatColor(hex: string): string {
            return hex.toUpperCase();
        }
    }
}

// ===== PROPERTIES & ACCESSORS =====
class ColorScheme {
    private _primary: string = "#000000";
    
    get primary(): string {
        return this._primary;
    }
    
    set primary(value: string) {
        this._primary = value;
    }
}

// ===== NULL & UNDEFINED =====
const nullValue = null;
const undefinedValue = undefined;
const optionalValue?: string = undefined;

// ===== SYMBOLS =====
const symbolKey = Symbol("key");
const wellKnownSymbol = Symbol.iterator;

// ===== OBJECTS & PROPERTIES =====
const themeConfig = {
    name: "Evergarden",
    version: 1.0,
    colors: {
        background: "#0E1012",
        foreground: "#e2e3e4"
    },
    ["computed-key"]: "value"
};

const { name, version } = themeConfig;
const { colors: { background, foreground } } = themeConfig;

// ===== ARRAY METHODS =====
const numbers = [1, 2, 3, 4, 5];
const doubled = numbers.map(n => n * 2);
const evens = numbers.filter(n => n % 2 === 0);
const sum = numbers.reduce((acc, n) => acc + n, 0);

// ===== SPREAD & REST =====
const array1 = [1, 2, 3];
const array2 = [...array1, 4, 5];

function restParams(...args: any[]) {
    console.log(args);
}

// ===== DESTRUCTURING =====
const [first, second, ...rest] = array1;
const { primary: mainColor, secondary: accentColor } = colorScheme;
